provider/pkg/gen/examples/upstream/generate.go

```diff
@@ -0,0 +1,234 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/contract"
+	"gopkg.in/yaml.v3"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+)
+
+//go:generate go run generate.go yaml .
+
+func main() {
+	if len(os.Args) < 3 {
+		fmt.Fprintf(os.Stderr, "Usage: %s <yaml source dir path> <markdown destination path>\n", os.Args[0])
+		os.Exit(1)
+	}
+	yamlPath := os.Args[1]
+	mdPath := os.Args[2]
+
+	if !filepath.IsAbs(yamlPath) {
+		cwd, err := os.Getwd()
+		contract.AssertNoError(err)
+		yamlPath = filepath.Join(cwd, yamlPath)
+	}
+
+	finfo, err := os.Lstat(mdPath)
+	if err != nil && os.IsNotExist(err) {
+		if err := os.MkdirAll(mdPath, 0600); err != nil {
+			panic(err)
+		}
+	}
+
+	if !finfo.IsDir() {
+		fmt.Fprintf(os.Stderr, "Expect markdown destination %q to be a directory\n", mdPath)
+		os.Exit(1)
+	}
+
+	yamls, err := os.ReadDir(yamlPath)
+	if err != nil {
+		panic(err)
+	}
+	for _, yamlFile := range yamls {
+		if err := processYaml(filepath.Join(yamlPath, yamlFile.Name()), mdPath); err != nil {
+			fmt.Fprintf(os.Stderr, "%+v", err)
+			os.Exit(1)
+		}
+	}
+}
+
+func processYaml(path string, mdDir string) error {
+	yamlFile, err := os.Open(path)
+	if err != nil {
+		return err
+	}
+
+	base := filepath.Base(path)
+	md := strings.NewReplacer(".yaml", ".md", ".yml", ".md").Replace(base)
+
+	buf := bytes.Buffer{}
+	_, err = buf.WriteString("{{% examples %}}\n")
```

Comment:
I think it would be easier to generate the markdown file using a template, but I won't block on this since the automation is already an improvement over the status quo.

Reply:
Yeah I thought about that. Definitely worth revisiting this with a template. There are still some rough edges with pulumi convert - as in it tries to install dependencies etc. which may fail. Mostly think of this as a script for now which can be run to update the examples as they are added. This is not even integrated into CI.